### PR TITLE
Retire `ChangesetCase`

### LINKED ISSUE #96: Retire `ChangesetCase`

### PR DESCRIPTION
**Pull request by indocomsoft**
Attempt to fix #96. Might be cancerous. Comments are welcome

### CODE REVIEW

File: test/support/changeset_helper.ex [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  ```
+  """
+
+  defmacro __using__(opt) do
+    if opt[:entity] do
+      quote do
+        alias Cadet.Repo
+
+        @spec test_changeset(map(), :assert | :refute, atom()) :: any()
+        defp test_changeset(params, assert_or_refute \\ :assert, function_name \\ :changeset) do
```
[the remark this answers is not in the document]
Comment (indocomsoft) [reply test/support/changeset_helper.ex]: Yeah, good idea

**Comment by indocomsoft** [comment]
@tuesmiddt Followed your suggestion

File: test/cadet/assessments/answer_test.exs [hunk 2]
```diff
@@ -115,22 +102,13 @@ defmodule Cadet.Assessments.AnswerTest do
...
-      {:ok, _} = Repo.delete(submission)
 
-      result =
-        %Answer{}
-        |> Answer.changeset(Map.put(params, :question_id, new_mcq_question.id))
-        |> Repo.insert()
+      {:ok, _} = Repo.delete(submission)
 
-      assert({:error, _} = result, inspect(result))
+      assert_changeset_db(Map.put(params, :question_id, new_mcq_question.id), :invalid)
```
Comment (tuesmiddt) [inline test/cadet/assessments/answer_test.exs]: prefer
```
params
|> Map.put(:question_id, new_mcq_question.id)
|> assert_changeset_db(:invalid)
```

File: test/cadet/course/announcement_test.exs [hunk 3]
```diff
@@ -1,13 +1,16 @@
...
+
+  describe "Changesets" do
+    test "valid changesets" do
+      assert_changeset(%{title: "title", content: "Hello world", published: true}, :valid)
+    end
 
-  invalid_changesets Announcement do
-    %{title: "", content: "Some content"}
+    test "invalid changeset" do
+      assert_changeset_db(%{title: "", content: "Some content"}, :invalid)
```
Comment (tuesmiddt) [inline test/cadet/course/announcement_test.exs]: This shouldn't be a db test

**Pull request merged**